fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/ClientRepositoryWrapper.java

```diff
@@ -44,6 +44,7 @@ public ClientRepositoryWrapper(final ClientRepository repository, final Platform
         this.context = context;
     }
 
+    @Transactional
```

Comment:
Are the "@Transactional" annotations really needed here? This is technically not solving the Sonar issues... I'd remove them for now... and if there's a reason why to put them then please add them to another PR

Reply:
Self-invocation cannot make @Transactional to take effect. @Transactional tells spring to create a proxy around the object. The proxy intercepts calls to the object from other objects, but it does not intercept calls within the object. And nested @Transactional is not a problem, because the default option is REQUIRED, so if `findOneWithNotFoundDetection(final Long id)` method has already opened a transaction, then `findOneWithNotFoundDetection(final Long clientId, final boolean loadLazyCollections)` won't be open a new one.